ourjapanlife/findadoc-server
# fix: createFacilityWithHealthcareProfessional and add tests

This PR is part of a 🥞
1. https://github.com/ourjapanlife/findadoc-server/pull/292 👈🏼 You are here
2. https://github.com/ourjapanlife/findadoc-server/pull/293 

# What changed
I ran into a lot of errors while attempting to write the happy path test for `createFacilityWithHealthcareProfessional`. Most of the errors were a result of passing the regular types instead of the inputs (Ex. Facility instead of FacilityInput). The other functions in `resolvers.ts` will likely need to be updated in future PRs to do the same.

Here's a breakdown of the changes:
1. Fixes the required fields for the schema fields
2. Changes the `createFacilityWithHealthcareProfessional` resolver function and related service functions to accept `Input` types
3. Updates the seed file types
4. Removes unneeded console.logs
5. Adds end-to-end tests for facilities in facilities.test.ts
6. Renames submissions.test.ts (previously submissionsService.test.ts)
7. Adds doc strings to some of the facility functions

# Testing instructions
1. Run `yarn test:dockerstart` to start the Docker container
2. Run `yarn test` and ensure the tests pass
3. Run `yarn dev` and try the following query in GraphQL studio:


**Operation:**
```gql
mutation Mutation($input: FacilityInput) {
  createFacilityWithHealthcareProfessional(input: $input) {
    contact {
      address {
        addressLine1En
        addressLine1Ja
        addressLine2En
        addressLine2Ja
        cityEn
        cityJa
        postalCode
        prefectureEn
        prefectureJa
      }
      email
      mapsLink
      phone
      website
    }
    healthcareProfessionalIds
    id
    isDeleted
    nameEn
    nameJa
    createdDate
    updatedDate
  }
}
```
**Variables:**
```json
{
  "input": {
    "contact": {
      "address": {
        "addressLine1Ja": "some address line",
        "addressLine1En": "some address line",
        "addressLine2En": "some address line",
        "addressLine2Ja": "some address line",
        "cityEn": "some city",
        "cityJa": "some city",
        "postalCode": "some postal code",
        "prefectureEn": "some prefecture",
        "prefectureJa": "some prefecture"
      },
      "email": "some@email.com",
      "mapsLink": "https://some-map-link",
      "phone": "000-000-0000",
      "website": "https://foo.com"
    },
    "healthcareProfessionalIds": [],
    "healthcareProfessionals": [
      {
        "acceptedInsurance": "JAPANESE_HEALTH_INSURANCE",
        "degrees": [
          {
            "nameJa": "some degree",
            "nameEn": "some degree",
            "abbreviation": "some degree"
          }
        ],
        "names": [
          {
            "middleName": "some name",
            "locale": "ENGLISH",
            "lastName": "some name",
            "firstName": "some name"
          }
        ],
        "specialties": [
          {
            "names": [
              {
                "name": "some specialty",
                "locale": "ENGLISH"
              }
            ]
          }
        ],
        "spokenLanguages": [
          {
            "nameNative": "some language",
            "nameJa": "some language",
            "nameEn": "some language",
            "iso639_3": "en"
          }
        ]
      }
    ],
    "nameEn": "some facility name",
    "nameJa": "some facility name"
  }
}
```

### Screenshot of Response
![Screenshot 2023-10-01 at 12 41 16 AM](https://github.com/ourjapanlife/findadoc-server/assets/31802656/541b78a6-69c8-401b-9b98-97dd2819fb42)
